.ci/lib.sh

```diff
@@ -59,3 +59,34 @@ function get_cc_versions(){
 	[ ! -f "$_versions_file" ] && { echo >&2 "ERROR: cannot find $_versions_file"; exit 1; }
 	source "$_versions_file"
 }
+
+
+function apply_depends_on() {
+	pushd "${GOPATH}/src/${kata_repo}"
+	label_lines=$(git log --format=%s%b -n 1 | grep "Depends-on:" || true)
```

Comment:
I wonder if we really need this restriction? I've been caught out with this before as my `Depends-on:` wasn't the top commit. Couldn't we just consider all the commits:

```
git log --format=%s%b master..
```

If we do that, we should probably check to ensure we only find a single `Depends-on:` though and error if we find more.

Reply:
@jodh-intel 
If we consider all commits, we may find `Depends-on:` from older PRs.

For finding a single `Depends-on:` I think there could be a case when we need to add more than one.
e.g.

```
Depends-on: github.com/kata-containers/agent#111
Depends-on: github.com/kata-containers/tests#222
```

Reply:
Sure, but if we use that `git log` command above, it will only look at the commits that are in the current (PR) branch but which are not in `master`, so it will be "constrained" to only look at relevant commits.

Yes, I guess if we do add a check, it should just check that there aren't `>1` `Depends-on:` for any one repository.

Reply:
Ohh, true, I tried your command without the dots, sorry about that. Agree, modifying...

Reply:
fixed.